Documentation.md

```diff
@@ -164,590 +174,606 @@ and made available to your JS client code.
+var queueArn = "arn:aws:connect:<REGION>:<ACCOUNT_ID>:instance/<CONNECT_INSTANCE_ID>/queue/<CONNECT_QUEUE_ID>";
 
 agent.connect(endpoint, {
   queueARN: queueArn,
-  success: function() { console.log("outbound call connected") },
+  success: function() { console.log("outbound call connected"); },
   failure: function(err) {
     console.log("outbound call connection failed");
     console.log(err);
   }
 });
 ```
+Creates an outbound contact to the given endpoint. You can optionally provide a `queueARN` to associate the contact with a queue.
 
-### `agent.getAllQueueARNs`
-```
-const ARNs = agent.getAllQueueARNs;
+Optional success and failure callbacks can be provided to determine if the operation was successful.
+
+### `agent.getAllQueueARNs()`
+```js
+var ARNs = agent.getAllQueueARNs();
 ```
-Returns a list of the ARNs associated with this agent's routing profile's queues. 
+Returns a list of the ARNs associated with this agent's routing profile's queues.
 
 ### `agent.getEndpoints` / `agent.getAddresses`
-```
-const endpoints = agent.getEndpoints(
-   queue_array,
+```js
+var queuesARNs = agent.getAllQueueARNs();
+var endpoint1 = connect.Endpoint.byPhoneNumber("+18005550101");
+var endpoint2 = connect.Endpoint.byPhoneNumber("+18005550102");
+agent.getEndpoints(
+   queuesARNs,
    {
-      succes: function_1,
-      failure: function_2
+      success: function(data) {
+         var endpoints = data.endpoints; // or data.addresses
+      },
+      failure: function(err) {
+      }
    },
    {
-      endpoints: [endpoint1, endpoint2, etc.]
+      endpoints: [endpoint, endpoint2]
       maxResults: 95
-); 
+   }
+);
 ```
 Returns the endpoints associated with the queueARNs specified in `queueARNs`, along with any endpoints specified in `pageInfoIn.endpoints`.
-* `queueARNs`: Required. Can be a signle QueueARN or a list of QueueARNs associated with the desired queues.
+* `queueARNs`: Required. Can be a single Queue ARN or a list of Queue ARNs associated with the desired queues.
 * `callbacks`: Optional. A structure containing success and failure handlers.
-   * `callbacks.success`: A function for handling a successful API call.
-   * `callbacks.failure`: A function for handling a failed API call.
-* `pageInfoIn`: Optional. A structure containing options for the call to the LARS CTI API getEndpoints. 
-   * `pageInfoIn.maxResults`: The maximum number of endpoints returned from this API.
-   * `pageInfoIn.endpoints`: A list of endpoints to add to the final list of endpoints returned by this API.
-   
+   * `success`: A function for handling a successful API call.
+   * `failure`: A function for handling a failed API call.
+* `pageInfoIn`: Optional. A structure containing options for the call to the API getEndpoints.
+   * `maxResults`: The maximum number of endpoints returned from this API.
+   * `endpoints`: A list of endpoints to add to the final list of endpoints returned by this API.
```

Comment:
I'm wondering whether this arguments were meant to be disclosed, of if it was an internal detail implementation (of the internal recursive pagination).

Reply:
Yes, sorry I didn't catch this yesterday. Can we get rid of lines 542->544?